datasets/flwr_datasets/federated_dataset_test.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2023 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Federated Dataset tests."""
+
+
+import unittest
+
+import pytest
+from parameterized import parameterized, parameterized_class
+
+import datasets
+from flwr_datasets.federated_dataset import FederatedDataset
+
+
+@parameterized_class(
+    [
+        {"dataset_name": "mnist", "test_split": "test"},
+        {"dataset_name": "cifar10", "test_split": "test"},
+        {"dataset_name": "fashion_mnist", "test_split": "test"},
+        {"dataset_name": "sasha/dog-food", "test_split": "test"},
+        {"dataset_name": "zh-plus/tiny-imagenet", "test_split": "valid"},
+    ]
```

Comment:
I extended the tests. I needed to add the test_split because the `zh-plus/tiny-imagenet` doesn't have the test split, but it has a valid split.